spring-boot/admin/src/test/java/eu/coatrack/admin/service/GatewayHealthMonitorServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package eu.coatrack.admin.service;
+
+import eu.coatrack.admin.model.repository.ProxyRepository;
+import eu.coatrack.api.Proxy;
+import eu.coatrack.api.ProxyStates;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.mockito.Mockito.when;
+
+public class GatewayHealthMonitorServiceTest {
+
+    @Mock
+    ProxyRepository proxyRepository;
+
+    @Mock
+    Proxy proxySample;
+
+    private List<Proxy> sampleProxies = new ArrayList<>();
+
+    @InjectMocks
+    private GatewayHealthMonitorService gatewayHealthMonitorService;
+
+    public Proxy createSampleProxy(boolean monitoredEnabled, String proxyName) {
+        Proxy proxy = new Proxy();
+        proxy.setName(proxyName);
+        proxy.setId("testIdFor-" + proxyName);
+        proxy.setMonitoringEnabled(monitoredEnabled);
+        proxy.updateTimeOfLastSuccessfulCallToAdmin_setToNow();
+        return proxy;
+    }
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        sampleProxies = Arrays.asList(createSampleProxy(true, "testProxyMonitoringEnabledA"), createSampleProxy(false, "testProxyMonitoringDisabledA"), proxySample);
```

Comment:
Very long line. I would insert a line break.

Reply:
I agree with you.
So consider it done